internal/pkg/gateway/endorsement.go

```diff
@@ -102,25 +109,41 @@ func (p *plan) update(endorser *endorser, endorsement *peer.ProposalResponse) []
 		}
 	}
 
+	// check the proposal responses are the same
+	if p.responsePayload == nil {
+		p.responsePayload = response.GetPayload()
+	} else {
+		if !bytes.Equal(p.responsePayload, response.GetPayload()) {
+			logger.Warnw("ProposalResponsePayloads do not match (base64)", "payload1", b64.StdEncoding.EncodeToString(p.responsePayload), "payload2", b64.StdEncoding.EncodeToString(response.GetPayload()))
+			return nil, errors.New("ProposalResponsePayloads do not match")
+		}
+	}
+
 	for i := p.nextLayout; i < len(p.layouts); i++ {
 		layout := p.layouts[i]
 		if layout == nil {
 			continue
 		}
 		if quantity, ok := layout.required[group]; ok {
+			// ensure endorser uniqueness
```

Comment:
How do you know that the certificate matches the layout you are currently trying? What if the certificate of the endorser actually matches a different layout?

Reply:
Moved the de-duplication code, as suggested below